### PR TITLE
fix: add side tx responses in vote and rename for clarity

### PR DESCRIPTION
**Pull request by jdkanani**

### CODE REVIEW

**Review by murthyvitwit — APPROVED** [review]
It looks good to me.

File: proto/tendermint/abci/types.proto [hunk 1]
```diff
@@ -419,33 +419,31 @@ service ABCIApplication {
...
 // Side channel requests
 //
 
 message RequestBeginSideBlock {
-  bytes hash = 1;
-  tendermint.types.Header header = 2 [(gogoproto.nullable) = false];  
-  repeated tendermint.types.SideTxResult side_tx_results = 3 [(gogoproto.nullable)=false];
+  bytes                   hash                              = 1;
+  tendermint.types.Header header                            = 2 [(gogoproto.nullable) = false];
+  repeated tendermint.types.SideTxResponses side_tx_results = 3 [(gogoproto.nullable) = false];
```
Comment (anilcse) [inline proto/tendermint/abci/types.proto:429]: Add gogoproto annotations for multi-word keys, ref: https://github.com/cosmos/cosmos-sdk/blob/v0.40.1/proto/cosmos/auth/v1beta1/auth.proto#L45

File: proto/tendermint/abci/types.proto [hunk 2]
```diff
@@ -419,33 +419,31 @@ service ABCIApplication {
...
 message ResponseDeliverSideTx {
-  uint32 code = 1;
-  string codespace = 2;
-  tendermint.types.SideTxResultType result = 4;
-  bytes data = 3;
+  uint32                            code      = 1;
+  string                            codespace = 2;
+  tendermint.types.SideTxResultType result    = 4;
+  bytes                             data      = 3;
 }
```
Comment (anilcse) [inline proto/tendermint/abci/types.proto:449]: missing formating

File: proto/tendermint/types/types.proto [hunk 3]
```diff
@@ -158,33 +160,42 @@ message TxProof {
...
-  NO = 2;
+  YES  = 1;
+  NO   = 2;
+}
+
+// Side tx responses
+message SideTxResult {
+  SideTxResultType result  = 1;
+  bytes            sig     = 2;
+  bytes            tx_hash = 3;
```
Comment (anilcse) [inline proto/tendermint/types/types.proto:178]: Should it be string?

File: proto/tendermint/types/types.proto [hunk 4]
```diff
@@ -158,33 +160,42 @@ message TxProof {
...
+//
 
-  bytes tx_hash = 1;
-  repeated SideTxSig sigs = 2 [(gogoproto.nullable)=false];
-    bytes data = 3;
+// Side-tx sig response
+message SideTxResponse {
+  SideTxResultType result  = 1;
+  bytes            sig     = 2;
+  bytes            address = 3;
```
Comment (anilcse) [inline proto/tendermint/types/types.proto:194]: should it be string?

**Pull request merged**